Why exclude datafusion-cli from workspace?

### Describe the bug

The tests for `datafusion-cli` is failing in the current main branch, but our CI doesn't find it.
```shell
$ cargo test
   Compiling datafusion-cli v26.0.0 (/Users/r4ntix/Workspace/github/arrow-datafusion/datafusion-cli)
error[E0277]: the `?` operator can only be applied to values that implement `Try`
   --> datafusion-cli/src/exec.rs:251:17
    |
251 |                 create_external_table(&ctx, cmd)?;
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `impl Future<Output = std::result::Result<(), datafusion::error::DataFusionError>>`
    |
    = help: the trait `Try` is not implemented for `impl Future<Output = std::result::Result<(), datafusion::error::DataFusionError>>`
help: consider `await`ing on the `Future`
    |
251 |                 create_external_table(&ctx, cmd).await?;
    |                                                 ++++++

For more information about this error, try `rustc --explain E0277`.
error: could not compile `datafusion-cli` due to previous error
```

I noticed that we currently exclude `datafusion-cli` from workspace.


### To Reproduce

```
$ cd datafusion-cli
$ cargo test
```

### Expected behavior

_No response_

### Additional context

_No response_

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -96,7 +96,10 @@ jobs:
         with:
           rust-version: stable
       - name: Run tests (excluding doctests)
-        run: cargo test --lib --tests --bins --features avro,json,dictionary_expressions
+        run: |
+          cargo test --lib --tests --bins --features avro,json,dictionary_expressions
+          cd datafusion-cli
+          cargo test --lib --tests --bins --all-features
       - name: Verify Working Directory Clean
         run: git diff --exit-code
 
@@ -152,7 +155,10 @@ jobs:
           rust-version: stable
       # Note: this does not include dictionary_expressions to reduce codegen
       - name: Run doctests
-        run: cargo test --doc --features avro,json
+        run: |
+          cargo test --doc --features avro,json
+          cd datafusion-cli
+          cargo test --doc --all-features
       - name: Verify Working Directory Clean
         run: git diff --exit-code
 
@@ -173,6 +179,8 @@ jobs:
         run: |
           export RUSTDOCFLAGS="-D warnings -A rustdoc::private-intra-doc-links"
           cargo doc --document-private-items --no-deps --workspace
+          cd datafusion-cli
+          cargo doc --document-private-items --no-deps
 
   # verify that the benchmark queries return the correct results
   verify-benchmark-results:
@@ -271,6 +279,8 @@ jobs:
         run: |
           export PATH=$PATH:$HOME/d/protoc/bin
           cargo test --lib --tests --bins --features avro,json,dictionary_expressions
+          cd datafusion-cli
+          cargo test --lib --tests --bins --all-features
         env:
           # do not produce debug symbols to keep memory usage down
           RUSTFLAGS: "-C debuginfo=0"
@@ -304,6 +314,8 @@ jobs:
         shell: bash
         run: |
           cargo test --lib --tests --bins --features avro,json,dictionary_expressions
+          cd datafusion-cli
+          cargo test --lib --tests --bins --all-features
         env:
           # do not produce debug symbols to keep memory usage down
           RUSTFLAGS: "-C debuginfo=0"

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -96,8 +96,32 @@ jobs:
         with:
           rust-version: stable
       - name: Run tests (excluding doctests)
+        run: cargo test --lib --tests --bins --features avro,json,dictionary_expressions
+      - name: Verify Working Directory Clean
+        run: git diff --exit-code
+
+  linux-test-datafusion-cli:
+    name: cargo test datafusion-cli (amd64)
+    needs: [ linux-build-lib ]
+    runs-on: ubuntu-latest
+    container:
+      image: amd64/rust
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Cache Cargo
+        uses: actions/cache@v3
+        with:
+          path: /github/home/.cargo
+          # this key equals the ones on `linux-build-lib` for re-use
+          key: cargo-cache-
+      - name: Setup Rust toolchain
+        uses: ./.github/actions/setup-builder
+        with:
+          rust-version: stable
+      - name: Run tests (excluding doctests)
         run: |
-          cargo test --lib --tests --bins --features avro,json,dictionary_expressions
           cd datafusion-cli
           cargo test --lib --tests --bins --all-features
       - name: Verify Working Directory Clean

--- a/ci/scripts/rust_clippy.sh
+++ b/ci/scripts/rust_clippy.sh
@@ -19,3 +19,5 @@
 
 set -ex
 cargo clippy --all-targets --workspace --features avro,pyarrow -- -D warnings
+cd datafusion-cli
+cargo clippy --all-targets --all-features -- -D warnings

--- a/ci/scripts/rust_docs.sh
+++ b/ci/scripts/rust_docs.sh
@@ -20,3 +20,5 @@
 set -ex
 export RUSTDOCFLAGS="-D warnings -A rustdoc::private-intra-doc-links"
 cargo doc --document-private-items --no-deps --workspace
+cd datafusion-cli
+cargo doc --document-private-items --no-deps

--- a/ci/scripts/rust_fmt.sh
+++ b/ci/scripts/rust_fmt.sh
@@ -19,3 +19,5 @@
 
 set -ex
 cargo fmt --all -- --check
+cd datafusion-cli
+cargo fmt --all -- --check

--- a/datafusion-cli/src/exec.rs
+++ b/datafusion-cli/src/exec.rs
@@ -313,7 +313,7 @@ mod tests {
         let err = create_external_table_test(location, &sql)
             .await
             .unwrap_err();
-        assert!(err.to_string().contains("No such file or directory"));
+        assert!(err.to_string().contains("os error 2"));
 
         // for service_account_key
         let sql = format!("CREATE EXTERNAL TABLE test STORED AS PARQUET OPTIONS('service_account_key' '{service_account_key}') LOCATION '{location}'");
@@ -328,14 +328,14 @@ mod tests {
         let err = create_external_table_test(location, &sql)
             .await
             .unwrap_err();
-        assert!(err.to_string().contains("No such file or directory"));
+        assert!(err.to_string().contains("os error 2"));
 
         Ok(())
     }
 
     #[tokio::test]
     async fn create_external_table_local_file() -> Result<()> {
-        let location = "/path/to/file.parquet";
+        let location = "path/to/file.parquet";
 
         // Ensure that local files are also registered
         let sql =
@@ -343,7 +343,12 @@ mod tests {
         let err = create_external_table_test(location, &sql)
             .await
             .unwrap_err();
-        assert!(err.to_string().contains("No such file or directory"));
+
+        if let DataFusionError::IoError(e) = err {
+            assert_eq!(e.kind(), std::io::ErrorKind::NotFound);
+        } else {
+            return Err(err);
+        }
 
         Ok(())
     }

--- a/datafusion-cli/src/exec.rs
+++ b/datafusion-cli/src/exec.rs
@@ -246,12 +246,13 @@ mod tests {
         let ctx = SessionContext::new();
         let plan = ctx.state().create_logical_plan(sql).await?;
 
-        match &plan {
-            LogicalPlan::Ddl(DdlStatement::CreateExternalTable(cmd)) => {
-                create_external_table(&ctx, cmd).await?;
-            }
-            _ => unreachable!(),
-        };
+        if let LogicalPlan::Ddl(DdlStatement::CreateExternalTable(cmd)) = &plan {
+            create_external_table(&ctx, cmd).await?;
+        } else {
+            return Err(DataFusionError::Plan(
+                "LogicalPlan is not a CreateExternalTable".to_string(),
+            ));
+        }
 
         ctx.runtime_env()
             .object_store(ListingTableUrl::parse(location)?)?;

--- a/datafusion-cli/src/object_storage.rs
+++ b/datafusion-cli/src/object_storage.rs
@@ -57,8 +57,7 @@ pub async fn get_s3_object_store_builder(
             .ok_or_else(|| {
                 DataFusionError::ObjectStore(object_store::Error::Generic {
                     store: "S3",
-                    source: "Failed to get S3 credentials from environment".to_string()
-                        .into(),
+                    source: "Failed to get S3 credentials from environment".into(),
                 })
             })?
             .clone();

--- a/datafusion-cli/src/print_format.rs
+++ b/datafusion-cli/src/print_format.rs
@@ -90,7 +90,6 @@ mod tests {
     use super::*;
     use arrow::array::Int32Array;
     use arrow::datatypes::{DataType, Field, Schema};
-    use datafusion::from_slice::FromSlice;
     use std::sync::Arc;
 
     #[test]
@@ -107,9 +106,9 @@ mod tests {
         let batch = RecordBatch::try_new(
             schema,
             vec![
-                Arc::new(Int32Array::from_slice([1, 2, 3])),
-                Arc::new(Int32Array::from_slice([4, 5, 6])),
-                Arc::new(Int32Array::from_slice([7, 8, 9])),
+                Arc::new(Int32Array::from(vec![1, 2, 3])),
+                Arc::new(Int32Array::from(vec![4, 5, 6])),
+                Arc::new(Int32Array::from(vec![7, 8, 9])),
             ],
         )
         .unwrap();
@@ -137,9 +136,9 @@ mod tests {
         let batch = RecordBatch::try_new(
             schema,
             vec![
-                Arc::new(Int32Array::from_slice([1, 2, 3])),
-                Arc::new(Int32Array::from_slice([4, 5, 6])),
-                Arc::new(Int32Array::from_slice([7, 8, 9])),
+                Arc::new(Int32Array::from(vec![1, 2, 3])),
+                Arc::new(Int32Array::from(vec![4, 5, 6])),
+                Arc::new(Int32Array::from(vec![7, 8, 9])),
             ],
         )
         .unwrap();

